packages/nouns-contracts/contracts/Rewards.sol

```diff
@@ -0,0 +1,591 @@
+        RewardsStorage storage $ = _getRewardsStorage();
+
+        uint256 numClientIds = nextTokenId();
+        uint32[] memory allClientIds = new uint32[](numClientIds);
+        for (uint32 i; i < numClientIds; ++i) {
+            allClientIds[i] = i;
+        }
+        NounsDAOTypes.ProposalForRewards[] memory proposals = nounsDAO.proposalDataForRewards(
+            $.nextProposalIdToReward,
+            lastProposalId,
+            allClientIds
+        );
+
+        uint32[] memory sumVotes = new uint32[](numClientIds);
+        for (uint256 i; i < proposals.length; ++i) {
+            for (uint256 j; j < numClientIds; ++j) {
+                sumVotes[j] += proposals[i].voteData[j].votes;
+            }
+        }
+
+        uint256 idx;
+        uint32[] memory nonZeroClientIds = new uint32[](numClientIds);
+        for (uint32 i; i < numClientIds; ++i) {
+            if (sumVotes[i] > 0) nonZeroClientIds[idx++] = i;
+        }
+
+        assembly {
+            mstore(nonZeroClientIds, idx)
+        }
+
+        return nonZeroClientIds;
+    }
+
+    function getAuctionRevenue(
+        uint256 firstNounId,
+        uint256 endTimestamp
+    ) public view returns (uint256 sumRevenue, uint256 lastAuctionId) {
+        INounsAuctionHouseV2.Settlement[] memory s = auctionHouse.getSettlementsFromIdtoTimestamp(
+            firstNounId,
+            endTimestamp,
+            true
+        );
+        sumRevenue = sumAuctions(s);
+        lastAuctionId = s[s.length - 1].nounId;
+    }
+
+    /**
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     *   PUBLIC READ - STORAGE GETTERS
+     * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
+     */
+
+    function nextAuctionIdToReward() public view returns (uint256) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.nextAuctionIdToReward;
+    }
+
+    function nextProposalIdToReward() public view returns (uint32) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.nextProposalIdToReward;
+    }
+
+    function nextProposalRewardFirstAuctionId() public view returns (uint256) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.nextProposalRewardFirstAuctionId;
+    }
+
+    function lastProposalRewardsUpdate() public view returns (uint256) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.lastProposalRewardsUpdate;
+    }
+
+    function getParams() public view returns (RewardParams memory) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.params;
+    }
+
+    function ethToken() public view returns (IERC20) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.ethToken;
+    }
+
+    function admin() public view returns (address) {
+        RewardsStorage storage $ = _getRewardsStorage();
+        return $.admin;
+    }
+
+    /**
+     * @notice Returns the raw value from _clientBalances mapping. Usually you want to use `clientBalance`
```

Comment:
worth expanding why we have this separation in this natspec, i.e. because of the 1 wei hack

Reply:
added